### PR TITLE
Added pyswitch wrapper to create port-channel, delete port-channel and retrieve list of port-channels

### PR DESCRIPTION
**Pull request by vijendarp**
Added LAG and ifXtable MIB's in SnmpMib.py
Added API's which convert ifname to id and ifid to ifname and return a dict of mappings
Create a new derived class for MLX specific MIBs.
Unit tested the pyswitch wrapper code. Need to integrate with NE action and test

### CODE REVIEW

File: pyswitch/snmp/mlx/base/interface.py [hunk 1]
```diff
@@ -105,4 +111,339 @@ def add_vlan_int(self, vlan_id_list, desc=None):
...
+            ifid_name = self.get_port_channel_member_ports(lag_name)
+            for item in ifid_name:
+                port = ifid_name[item]
+                port = port.replace('ethernet', 'ethernet ')
+                cli_arr.append('disable' + " " + str(port))
+
+            cli_arr.append('no' + " " + 'lag' + " " + lag_name)
+            self._callback(cli_arr, handler='cli-set')
+            return True
+        except Exception as error:
```
[the remark this answers is not in the document]
Comment (vijendarp) [reply pyswitch/snmp/mlx/base/interface.py]: Addressed the review comments. Please approve

[the remark this answers is not in the document]
Comment (vijendarp) [reply pyswitch/snmp/mlx/base/interface.py]: Also, integrated with NE actions for lag create and lag delete and tested the pyswitch wrapper code. NE action needed some changes which will be checked in separately.

File: pyswitch/snmp/mlx/base/interface.py [hunk 2]
```diff
@@ -105,4 +119,368 @@ def add_vlan_int(self, vlan_id_list, desc=None):
         except Exception as error:
             reason = error.message
             raise ValueError('Failed to create VLAN due to %s', reason)
-            return False
+
+    def create_port_channel(self, ports, int_type, portchannel_num, mode, desc=None):
+        """create port channel
```
Comment (dpasupathi) [inline pyswitch/snmp/mlx/base/interface.py]: @vijendarp : I hope the signature difference between SLX/VDX to MLX will not have much impact on NE action side?

Comment (vijendarp) [reply pyswitch/snmp/mlx/base/interface.py]: @dpasupathi : NE action changes are made and I don't see a big impact on it. Changes are confined to MLX/NI in NE action

File: pyswitch/snmp/base/interface.py [hunk 3]
```diff
@@ -86,15 +87,24 @@ def del_vlan_int(self, vlan_id):
...
         Raises:
-            None
+            InvalidVlanId, ValueError
+
+        Examples:
+            >>> import pyswitch.device
+            >>> conn = ('10.24.85.107', '22')
+            >>> auth = ('admin', 'admin')
+            >>> with pyswitch.device.Device(conn=conn, auth=auth) as dev:
+            ...     output = dev.interface.del_vlan_int(300)
```
Comment (samirsss) [inline pyswitch/snmp/base/interface.py]: Lets do a add_vlan before we do a delete

File: pyswitch/snmp/base/interface.py [hunk 4]
```diff
@@ -109,7 +119,28 @@ def del_vlan_int(self, vlan_id):
...
+
+        Raises:
+            InvalidVlanId
+
+        Examples:
+            >>> import pyswitch.device
+            >>> conn = ('10.24.85.107', '22')
+            >>> auth = ('admin', 'admin')
+            >>> with pyswitch.device.Device(conn=conn, auth=auth) as dev:
+            ...     output = dev.interface.get_vlan_int(300)
```
Comment (samirsss) [inline pyswitch/snmp/base/interface.py]: Create a vlan before we go a get. Do we check for true false? Is there an assert to make sure this test passes? https://github.com/StackStorm/PySwitchLib/blob/master/pyswitch/os/base/interface.py#L536

File: pyswitch/snmp/mlx/base/interface.py [hunk 5]
```diff
@@ -95,6 +101,14 @@ def add_vlan_int(self, vlan_id_list, desc=None):
...
         Raises:
             None
+
+        Examples:
+            >>> import pyswitch.device
+            >>> conn = ('10.24.85.107', '22')
+            >>> auth = ('admin', 'admin')
+            >>> with pyswitch.device.Device(conn=conn, auth=auth) as dev:
+            ...     vlan_list = [700,800, 900]
+            ...     output = dev.interface.add_vlan_int(vlan_list, 'vlan_name')
```
Comment (samirsss) [inline pyswitch/snmp/mlx/base/interface.py]: Similar comment as above.

File: pyswitch/snmp/mlx/base/interface.py [hunk 6]
```diff
@@ -105,4 +119,368 @@ def add_vlan_int(self, vlan_id_list, desc=None):
...
+        raises:
+            keyerror: if `int_type`, `name`, or `description` is not specified.
+            valueerror: if `name` or `int_type` are not valid values.
+
+        Examples:
+            >>> import pyswitch.device
+            >>> conn = ('10.24.85.107', '22')
+            >>> auth = ('admin', 'admin')
+            >>> with pyswitch.device.Device(conn=conn, auth=auth) as dev:
+            ...     output = dev.interface.remove_port_channel(20)
```
Comment (samirsss) [inline pyswitch/snmp/mlx/base/interface.py]: Can we add before we remove a PO? Also can we assert the output https://github.com/StackStorm/PySwitchLib/blob/master/pyswitch/os/base/interface.py#L536

File: pyswitch/snmp/mlx/base/interface.py [hunk 7]
```diff
@@ -105,4 +119,368 @@ def add_vlan_int(self, vlan_id_list, desc=None):
...
+        raises:
+            keyerror: if `int_type`, `name`, or `description` is not specified.
+            valueerror: if `name` or `int_type` are not valid values.
+
+        Examples:
+            >>> import pyswitch.device
+            >>> conn = ('10.24.85.107', '22')
+            >>> auth = ('admin', 'admin')
+            >>> with pyswitch.device.Device(conn=conn, auth=auth) as dev:
+            ...     output = dev.interface.port_channels
```
Comment (samirsss) [inline pyswitch/snmp/mlx/base/interface.py]: Add a PO and then assert that the length of the output  > =1

File: pyswitch/snmp/mlx/base/interface.py [hunk 8]
```diff
@@ -105,4 +119,368 @@ def add_vlan_int(self, vlan_id_list, desc=None):
...
+        raises:
+            keyerror: if `int_type`, `name`, or `description` is not specified.
+            valueerror: if `name` or `int_type` are not valid values.
+
+        Examples:
+            >>> import pyswitch.device
+            >>> conn = ('10.24.85.107', '22')
+            >>> auth = ('admin', 'admin')
+            >>> with pyswitch.device.Device(conn=conn, auth=auth) as dev:
+            ...     output = dev.interface.get_port_channel_member_ports('po50')
```
Comment (samirsss) [inline pyswitch/snmp/mlx/base/interface.py]: Lets add a PO and then get it and assert the output

**Pull request merged**